### PR TITLE
WW-349 start redlink flow

### PR DESCRIPTION
**Pull request by idradm**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [CreatePage.js#L335](https://github.com/Wikia/app/blob/a2da5ff1e7529428820e0ffd8f6431bda737f13e/extensions/wikia/CreatePage/js/CreatePage.js#L335): Remove this "===" check; it will always be false. Did you mean to use "=="? [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS3403)
2. CRITICAL [CreatePage.js#L335](https://github.com/Wikia/app/blob/a2da5ff1e7529428820e0ffd8f6431bda737f13e/extensions/wikia/CreatePage/js/CreatePage.js#L335): Change this condition so that it does not always evaluate to "false". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2583)
3. MAJOR [CreatePage.js#L39](https://github.com/Wikia/app/blob/a2da5ff1e7529428820e0ffd8f6431bda737f13e/extensions/wikia/CreatePage/js/CreatePage.js#L39): Function has a complexity of 14 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
4. MAJOR [CreatePage.js#L131](https://github.com/Wikia/app/blob/a2da5ff1e7529428820e0ffd8f6431bda737f13e/extensions/wikia/CreatePage/js/CreatePage.js#L131): Remove the unused function parameter "event". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
5. MAJOR [CreatePage.js#L201](https://github.com/Wikia/app/blob/a2da5ff1e7529428820e0ffd8f6431bda737f13e/extensions/wikia/CreatePage/js/CreatePage.js#L201): Restrict what this loop acts on by testing each property. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)
6. MINOR [CreatePage.js#L300](https://github.com/Wikia/app/blob/a2da5ff1e7529428820e0ffd8f6431bda737f13e/extensions/wikia/CreatePage/js/CreatePage.js#L300): Reformat the code to have only one statement per line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AOneStatementPerLine)

File: extensions/wikia/CreatePage/js/CreatePage.js [hunk 1]
```diff
@@ -339,9 +340,23 @@ var CreatePage = {
 					}
 				}
 			});
+		} else {
+			// in case create page modal is disabled
+			$( '#WikiaArticle' ).on( 'click', 'a.new', function( e ) {
+				CreatePage.trackCreatePageStart();
+				e.preventDefault();
```
Comment (sebastianmarzjan) [inline extensions/wikia/CreatePage/js/CreatePage.js:347]: Won't that break open in new tab functionality ?

File: resources/wikia/modules/flowTracking.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
 define('wikia.flowTracking', ['wikia.tracker', 'wikia.window', 'mw', 'jquery'], function (tracker, w, mw, $) {
 
 	var flows = {
-			CREATE_PAGE_DIRECT_URL: 'create-page-direct-url'
+			CREATE_PAGE_DIRECT_URL: 'create-page-direct-url',
+			CREATE_PAGE_REDLINK: 'create-page-redlink'
```
Comment (sebastianmarzjan) [inline resources/wikia/modules/flowTracking.js:5]: I propose to use:

```
CREATE_PAGE_ARTICLE_REDLINK: 'create-page-article-redlink'
```

File: extensions/wikia/CreatePage/js/CreatePage.js [hunk 3]
```diff
@@ -339,9 +340,23 @@ var CreatePage = {
 					}
 				}
 			});
+		} else {
+			// in case create page modal is disabled
+			$( '#WikiaArticle' ).on( 'click', 'a.new', function( e ) {
```
Comment (sebastianmarzjan) [inline extensions/wikia/CreatePage/js/CreatePage.js:345]: Is this selector precise enough?

Comment (rybmat) [reply extensions/wikia/CreatePage/js/CreatePage.js:345]: Having WW-372 and WW-373 tickets in backlog, is it possible to distinguish between redlinks on special pages and redlingks on other pages?

File: resources/wikia/modules/flowTracking.js (hunk 2 above)
Comment (rybmat) [reply resources/wikia/modules/flowTracking.js:5]: I don't think that current implementation distinguishes between article and non-article redlinks

Comment (sebastianmarzjan) [reply resources/wikia/modules/flowTracking.js:5]: Yeah, I mean that we need the ability to distinguish between the two

**Comment by rybmat** [comment]
merged into WW-343

**Pull request closed without merging**